bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/teaser/CommerceTeaserActionItemImpl.java

```diff
@@ -15,19 +15,28 @@
 package com.adobe.cq.commerce.core.components.internal.models.v1.teaser;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
-import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.adobe.cq.commerce.core.components.models.teaser.CommerceTeaserActionItem;
 
-public class CommerceTeaserActionItem implements ListItem {
+public class CommerceTeaserActionItemImpl implements CommerceTeaserActionItem {
 
     private String title;
     private String url;
+    private String categoryId;
+    private String productSlug;
 
-    public CommerceTeaserActionItem(String title, String url) {
+    public CommerceTeaserActionItemImpl(String title, String url) {
         this.title = title;
         this.url = url;
     }
 
+    public CommerceTeaserActionItemImpl(String title, String url, String categoryId, String productSlug) {
```

Comment:
I'm a bit concerned that we hardcode the identifier types, that is, `id` for category and `slug` for product. I think the implementation in `CommerceTeaserImpl` was always pretty much "hardcoding" the fact that we use ids and slugs, but we should maybe avoid hardcoding this any further. Nevertheless, if we want to keep these ids, we should make sure we define some "generic" names that we can use in all components. We should avoid that one component for example exports `productSlug` while another exports `url_key`, etc, maybe we should discuss if we want to define general identifier names or reuse the magento field names.

Reply:
I understand your concern - if tomorrow we decide that the category picker should return a different field than `categoryId`, then this parameter name becomes invalid. The best we can do today is replace the names of the methods to be something generic like `getCategoryIdentifier()` and `getProductIdentifier()` 

> We should avoid that one component for example exports productSlug while another exports url_key

But these two represent two different things from the backend. In this case, the component exports both, depending on what is configured - a slug for the product or an URL key for the category. We chose these two because it's easier to work with them. Otherwise, we could've just used "entity id" and do the extra query to get the actual data that we need.

> we should discuss if we want to define general identifier names or reuse the Magento field names.

What would we use instead? We currently identify products by "SKU" and by "slug" (not sure if we use the product id anywhere). We can rename these to something else, but I don't see any point in doing it. The same for categories - we use the id and the url_key depending on what situation we're in.

Reply:
I think you're slightly confusing some fields: what we call "slug" is the product `url_key`, for categories we can use the `url_path` in the URL but only for SEO, not to lookup the category. So for products `slug = url_key`.

In `UrlProvider` we have enums for the id types, so maybe we could rather export something like:
```
productIdentifier: {
  type: SKU | URL_KEY,
  value: xyz
}
```
but it could be something that we use for all components, so we have a consistent way of consuming this in all client-side components. I don't want to be a pain, but it would be nice if we first "design" the export types in a consistent way, to avoid that client-side code has to change or stops working if, for example, a component is configured differently to use skus or slugs.

Reply:
> so maybe we could rather export something like [...]

So the client-side component would use different queries (or rather, different filters for the query) depending on what type of "identifier" it gets from AEM, did I get this correct? 

>to avoid that client-side code has to change or stops working if, for example, a component is configured differently to use SKUs or slugs.

This is a bit unclear to me - are you talking about _the same_ component, i.e. if someone updates the product field in the Teaser component to return an SKU instead of a slug?

Reply:
>So the client-side component would use different queries (or rather, different filters for the query) depending on what type of "identifier" it gets from AEM, did I get this correct?

Yes this is what we already do server-side for the product component at https://github.com/adobe/aem-core-cif-components/blob/master/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/product/ProductImpl.java#L139

> This is a bit unclear to me - are you talking about the same component, i.e. if someone updates the product field in the Teaser component to return an SKU instead of a slug?

Yes exactly. One problem is that the picker can be changed to set the SKU or SLUG, but most components do not support this (except the product component). I think usually they expect the SKU, but as you see in your PR, the commerce teaser uses the SLUG, so that's why I think we should make sure we don't introduce another level of technical debt where client-side code has "hardcoded" identifier types.

To me this is really something we didn't do well (in the picker and components), so we should probably invest a bit of time in designing this properly to avoid introducing more technical debt in this area.